include/ignition/utils/ImplPtr.hh

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright (C) 2018 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef IGNITION_UTILS_IMPLPTR_HH_
+#define IGNITION_UTILS_IMPLPTR_HH_
+
+#include <memory>
+#include <utility>
+
+#include <ignition/utils/detail/DefaultOps.hh>
+
+namespace ignition
+{
+  namespace utilities
```

Comment:
Can we have the namespace match the package / directory? i.e. `utils`. I don't want it to collide with the stuff in `ign-cmake`...